client/src/components/TeacherRegistration.jsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+
+const handleSubmit = async (e) => {
+  e.preventDefault();
+
+  // const data = {
+  //   e.target.
+  //   username: document.getElementById("username").value,
+  //   password: document.getElementById("password").value,
+  // };
+
+  if (
+    data.teacherid != null &&
+    data.firstname != null &&
+    data.lastname != null &&
+    data.password != null
+  ) {
+    // Ensure none of the input fields are empty
+    if (
+      data.teacherid.trim() !== "" &&
+      data.firstname.trim() !== "" &&
+      data.lastname.trim() !== "" &&
+      data.password.trim() !== ""
+    ) {
+      const response = await fetch("/teacherreg", {
+        // Change to "/teacherreg"
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(data),
+      });
+    } else {
+      console.error("One or more fields are empty");
+    }
+  } else {
+    console.error("One or more fields are null");
+  }
+
+  const response = await fetch("/teacherreg", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data),
+  });
```

Comment:
```suggestion
```
I think this should be removed, you are posting twice, once if the info is there, if the info is missing? we should only post if the info is there